### PR TITLE
Armor material builder, item and block translation keys. 

### LINKED ISSUE #37: set the translation key for blocks and items while registering

### LINKED ISSUE #38: make an armor material builder

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
- Automatically sets the translation keys for items and blocks. 
- Added a builder for armor materials
- Did something to the test mod

Closes #37 
Closes #38

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ plugins {
 def ENV = System.getenv()
 
 class Globals {
-	static def baseVersion = "0.7.0"
+	static def baseVersion = "0.7.1"
```
Comment (hYdos) [inline build.gradle:18]: i just pushed a version called 0.7.1 this wont work

**Review by hYdos — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 2]
```diff
@@ -15,7 +15,7 @@ plugins {
 def ENV = System.getenv()
 
 class Globals {
-	static def baseVersion = "0.7.1"
+	static def baseVersion = "0.7.2"
```
Comment (hYdos) [inline build.gradle:18]: another newer version is built

**Review by hYdos — APPROVED** [review]

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/BlockRegistry.java [hunk 3]
```diff
@@ -28,6 +28,7 @@ private BlockRegistry() {
 	}
 
 	public static <T extends Block> T register(Identifier id, T block) {
+		block.setTranslationKey(id.toString());
```
Comment (shedaniel) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/BlockRegistry.java:31]: is this correct? isn't this some weird tile.modid.id.name

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/RarityProvider.java [hunk 4]
```diff
@@ -19,14 +19,11 @@
...
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.Formatting;
-import net.minecraft.util.Identifier;
 
 /**
  * Allows giving an Item a custom rarity. This specifies the color of the Item's name when held or when hovered over.
  * This will override the value returned in {@link Item#getRarity(ItemStack)}.
  */
 public interface RarityProvider {
-	Identifier getId(ItemStack stack);
```
Comment (shedaniel) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/RarityProvider.java:29]: turn this into a default method instead

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+package net.fabricmc.fabric.api.object.builder.v1.block;
+
+import java.util.function.Predicate;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.armor.v1.ArmorMaterial;
+
+public class ArmorMaterialBuilder {
```
Comment (shedaniel) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java:26]: I see no reason for this class

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/RarityProvider.java (hunk 4 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/RarityProvider.java:29]: concern

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/BlockRegistry.java (hunk 3 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/BlockRegistry.java:31]: that gets done automatically in `getTranslationKey`
`setTranslationKey` is only the id
in minecraft names like `notGate` are used

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java (hunk 5 above)
Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java:26]: i do

Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java:26]: it makes building them easier

Comment (shedaniel) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/ArmorMaterialBuilder.java:26]: There is no point if you are not providing anything special in the builder

**Pull request merged**